oak/server/rust/oak_runtime/src/node/wasm.rs

```diff
@@ -877,35 +877,94 @@ impl super::Node for WasmNode {
     }
 }
 
-#[test]
-fn wasm_invalid_args() {
-    use super::Node;
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::node::Node;
     use crate::runtime::TEST_NODE_ID;
-    let configuration = crate::runtime::Configuration {
-        nodes: HashMap::new(),
-        entry_module: "test_module".to_string(),
-        entrypoint: "test_function".to_string(),
-    };
-    let runtime_ref = crate::runtime::Runtime::create(configuration).into_ref();
-    let (_, reader_handle) =
-        runtime_ref.new_channel(TEST_NODE_ID, &oak_abi::label::Label::public_trusted());
-
-    // From https://docs.rs/wasmi/0.6.2/wasmi/struct.Module.html#method.from_buffer:
-    // Minimal module:
-    // \0asm - magic
-    // 0x01 - version (in little-endian)
-    let module =
-        wasmi::Module::from_buffer(vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00]).unwrap();
-
-    let mut node = WasmNode::new(
-        "test",
-        runtime_ref,
-        TEST_NODE_ID,
-        Arc::new(module),
-        "test_function".to_string(),
-        reader_handle,
-    );
-
-    // The minimal module does not have the necessary entry point, so it should fail immediately.
-    assert_eq!(Err(OakStatus::ErrInvalidArgs), node.start());
+    use wat::{parse_file, parse_str};
+
+    fn setup_node<S: AsRef<[u8]>>(buffer: S, entrypoint: &str) -> Box<dyn Node> {
+        let configuration = crate::runtime::Configuration {
+            nodes: HashMap::new(),
+            entry_module: "test_module".to_string(),
+            entrypoint: entrypoint.to_string(),
+        };
+        let runtime_ref = crate::runtime::Runtime::create(configuration).into_ref();
+        let (_, reader_handle) =
+            runtime_ref.new_channel(TEST_NODE_ID, &oak_abi::label::Label::public_trusted());
+
+        let module = wasmi::Module::from_buffer(buffer).unwrap();
+
+        Box::new(WasmNode::new(
+            "test",
+            runtime_ref,
+            TEST_NODE_ID,
+            Arc::new(module),
+            entrypoint.to_string(),
+            reader_handle,
+        ))
+    }
+
+    fn from_testfile(path: &str) -> Vec<u8> {
+        parse_file(path).unwrap()
+    }
+
+    #[test]
+    fn wasm_starting_module_without_content_fails() {
+        // Loads an empty module, equivalent to: vec![0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00,
```

Comment:
May as well leave around an actual test case with that vec, and here just mention that this tests corresponds to that. Or perhaps compare binary with that vec. Basically prove what you are stating in the comment, but as code :)